src/main/java/eu/dissco/backend/service/SpecimenService.java

```diff
@@ -287,18 +287,30 @@ private Map<String, List<String>> mapParams(MultiValueMap<String, String> params
 
   public JsonApiWrapper aggregations(MultiValueMap<String, String> params, String path)
       throws IOException, UnknownParameterException {
-    var aggregations = elasticRepository.getAggregations(mapParams(params));
-    var dataNode = new JsonApiData(String.valueOf(params.hashCode()), "aggregations",
+    var aggregations = elasticRepository.getAggregations(mapParamsKeyword(params));
+    var dataNode = new JsonApiData(String.valueOf(params.hashCode()), AGGREGATIONS_TYPE,
         mapper.valueToTree(aggregations));
     return new JsonApiWrapper(dataNode, new JsonApiLinks(path));
   }
 
   public JsonApiWrapper discipline(String path) throws IOException {
     var disciplineResult = elasticRepository.getAggregation(TOPIC_DISCIPLINE);
-    var dataNode = new JsonApiData(String.valueOf(path.hashCode()), "aggregations",
+    var dataNode = new JsonApiData(String.valueOf(path.hashCode()), AGGREGATIONS_TYPE,
         mapper.valueToTree(disciplineResult.getRight()));
     return new JsonApiWrapper(dataNode,
         new JsonApiLinks(path), new JsonApiMeta(disciplineResult.getLeft()));
   }
 
+  public JsonApiWrapper searchTermValue(String name, String value, String path)
+      throws UnknownParameterException, IOException {
+    var mappedTerm = getMappedTerm(name);
+    if (mappedTerm.isPresent()){
+      var aggregations = elasticRepository.searchTermValue(name, mappedTerm.get(), value);
```

Comment:
Wouldn't name and mappedTerm.get() always be the same?

Reply:
No, the getMappedTerm get's the fullName/elasticName for the field. In the MappingTerms there is a map with the name and fullName. In a few cases they are the same, for example `midsLevel`